### PR TITLE
[test] Updated Introduction Tutorials & Add Automated Notebook Testing #38

### LINKED ISSUE #38: [Roadmap]: Test & Update Introduction Tutorials Given New Core Narwhals Backend Updates
**Is your request related to a feature in the roadmap? Please specify.**
Related to the new Narwhals backend integration in core_utils.py and target_shifter.py, affecting all introduction tutorials and dataset usage.

**Describe the problem or enhancement you'd like to address**
Need to test and update all introduction tutorials to use:
1. New Narwhals backend implementation
2. DatasetLoader for consistent data loading
3. Healthcare dataset integration following standard design:
   - If standard academic dataset: Add to DatasetLoader
   - If non-standard dataset: Use core TemporalDataLoader with consistent design pattern

Tutorials to update:
1. tutorial_notebooks/introduction/0_narhwals_backend.ipynb
2. tutorial_notebooks/introduction/1_load_data_timeframe.ipynb
3. tutorial_notebooks/introduction/2_sliding_window_target_shifter.ipynb
4. tutorial_notebooks/introduction/3_target_shifter.ipynb
5. tutorial_notebooks/3_health_monitoring_analysis.ipynb (ensure consistent design pattern)

**Describe the solution you'd like**
1. Update tutorials to use consistent data loading:
```python
# For standard datasets
from temporalscope.datasets.datasets import DatasetLoader
loader = DatasetLoader("macrodata")
data = loader.load_data(backend="polars")

# For non-standard datasets (e.g., healthcare)
from temporalscope.core.temporal_data_loader import TimeFrame
# Follow same design pattern as other tutorials
```

2. Test each tutorial:
- Run with core backends
- Use consistent data loading patterns
- Fix any compatibility issues
- Update examples

**Describe alternatives you've considered**
1. Keep current data loading (inconsistent across tutorials)
2. Only update backend code (misses dataset standardization)
3. Different patterns for different datasets (reduces maintainability)

**Additional context**
Related files:
- src/temporalscope/core/core_utils.py
- src/temporalscope/target_shifters/single_step.py
- src/temporalscope/datasets/datasets.py
- All tutorials in tutorial_notebooks/

### PR DESCRIPTION
**Pull request by philip-ndikum**
# Pull Request

## PR Type
- [x] 🧪 Test Addition/Update
- [x] ♻️ Refactoring
- [ ] Other

## Description
This PR implements comprehensive updates to the introduction tutorials following the new Core Narwhals backend changes. Key improvements include:

1. **Tutorial Updates**:
   - Replaced deprecated temporal_target_shifter with new SingleStepTargetShifter
   - Added exploration cells for better data examination
   - Updated documentation to match core engineering design
   - Improved synthetic health monitoring tutorial with academic dataset roadmap

2. **Automated Testing**:
   - Integrated papermill for automated notebook testing
   - Added test infrastructure to verify notebook execution
   - Ensures tutorials remain functional across environment changes

3. **Code Organization**:
   - Renamed tutorials for clarity (e.g., single_step_target_shifting)
   - Removed redundant target_shifter notebook
   - Added clear implementation notes and future plans

## Related Issues
- Closes #38 ([Roadmap]: Test & Update Introduction Tutorials Given New Core Narwhals Backend Updates)

### Added/Updated Tests
- [x] Yes
- [ ] No

## How Has This Been Tested?
- [x] Unit: Added papermill-based notebook execution tests
- [x] Integration: Verified tutorials work across different backends
- [x] Manual: Tested notebook cells and data exploration features

### Test Configuration
- Python version: 3.11.10
- Operating System: macOS Sonoma
- Dependencies: papermill, jupyter, narwhals core

## Checklist
- [x] I have read the [contributors guidelines](https://github.com/philip-ndikum/TemporalScope/blob/main/CONTRIBUTING.md)
- [x] I have updated the documentation accordingly
- [x] I have added/updated tests (papermill integration)
- [x] I have updated issue #38 with implementation details
- [x] I have used conventional commits
- [x] I have run pre-commit hooks and fixed all issues

## Additional Notes
The automated notebook testing ensures tutorials remain functional as the codebase evolves. This infrastructure will be particularly valuable when we integrate academic datasets in future releases.

### CODE REVIEW

**Comment by kanenorman** [comment]
@philip-ndikum - LGTM.

File: test/integration/test_notebooks.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import pytest
+
+
+def get_notebooks():
+    """Get all notebooks from tutorial directory."""
+    notebook_dir = Path("tutorial_notebooks")
+    return list(notebook_dir.rglob("*.ipynb"))
+
+
+@pytest.mark.parametrize("notebook_path", get_notebooks())
```
[the remark this answers is not in the document]
Comment (philip-ndikum) [reply test/integration/test_notebooks.py:32]: Ah yes let me try & do that ASAP - thank you so much!

[the remark this answers is not in the document]
Comment (kanenorman) [reply test/integration/test_notebooks.py:32]: will also need to modify `pyproject.toml`

I don't remember the syntax, but something like this I believe.

```toml
[tool.hatch.envs.test.scripts]
unit = 'pytest --cov-report=lcov --cov-report=term --cov="temporalscope" -m "not integration and not notebook" {args:test}'
```

[the remark this answers is not in the document]
Comment (philip-ndikum) [reply test/integration/test_notebooks.py:44]: Added changes to address issues

In **pyproject.toml**:

`
[tool.hatch.envs.test.scripts]
unit = 'pytest --cov-report=lcov --cov-report=term --cov="temporalscope" -m "not integration and not notebook" {args:test}'
notebook = 'pytest --cov-report=lcov --cov-report=term --cov="temporalscope" -m "notebook" {args:test}'
`

Then in **test_notebooks.py**:

`
@pytest.mark.notebook
@pytest.mark.parametrize("notebook_path", get_notebooks())
def test_notebook_runs(notebook_path, tmp_path):
    """Test that notebook executes without errors using Papermill.
    
    Uses the notebook marker to separate execution tests from unit tests,
    focusing solely on verifying that notebooks can run successfully.
    """
`

Thank you very much

**Review by kanenorman — APPROVED** [review]
Made minor edits. LGTM!

**Pull request merged**